environments/servlet/core/src/main/java/org/jboss/weld/environment/jetty/JettyContainer.java

```diff
@@ -62,21 +59,14 @@ protected String classToCheck() {
     @Override
     public boolean touch(ResourceLoader resourceLoader, ContainerContext context) throws Exception {
         ServletContext servletContext = context.getServletContext();
-        // The jetty cdi modules from 9.4.20 sets JETTY_CDI_ATTRIBUTE to indicate that a
-        // DecoratingListener is registered. Weld-3.1.2.Final documented that the decorate module
-        // could be used directly, which sets DECORATING_LISTENER_ATTRIBUTE
-        return servletContext.getAttribute(JETTY_CDI_ATTRIBUTE) instanceof String ||
-                servletContext.getAttribute(DECORATING_LISTENER_ATTRIBUTE) instanceof String;
+        return servletContext.getAttribute(JETTY_CDI_ATTRIBUTE) instanceof String;
     }
 
     @Override
     public void initialize(ContainerContext context) {
         try {
             ServletContext servletContext = context.getServletContext();
             String mode = (String) servletContext.getAttribute(JETTY_CDI_ATTRIBUTE);
-            if (mode == null) {
-                mode = DECORATING_LISTENER_MODE;
-            }
             switch (mode) {
                 case CDI_SPI_DECORATOR_MODE:
```

Comment:
According to https://github.com/arquillian/arquillian-container-jetty/issues/197#issuecomment-2077383639, this is Jetty's internal class. I suppose the only purpose it serves here is telling apart an undefined state (i.e. throwing an exception) versus this state which, for us, is just a no-op.
I'll add a code comment for my future self to remember that.

Reply:
I've added a code comment clarification.